packages/hoprd/src/api/v2.ts

```diff
@@ -121,11 +121,7 @@ export async function setupRestApi(
         // Applying multiple URI encoding is an identity
         let apiTokenFromUser = encodeURIComponent(req.get('x-auth-token') || '')
 
-        if (
-          !this.options.disableApiAuthentication &&
-          this.options.apiToken !== undefined &&
-          apiTokenFromUser !== encodedApiToken
-        ) {
+        if (this.options.apiToken !== undefined && apiTokenFromUser !== encodedApiToken) {
```

Comment:
The `!this.options.disableApiAuthentication` clauses were redundant, because there `if (this.options.disableApiAuthentication) return true` on L119 and L135 respectively.